sdk/nodejs/output.ts

```diff
@@ -512,39 +567,36 @@ export function all<T1, T2, T3>(values: [Input<T1> | undefined, Input<T2> | unde
 export function all<T1, T2>(values: [Input<T1> | undefined, Input<T2> | undefined]): Output<[Unwrap<T1>, Unwrap<T2>]>;
 export function all<T>(ds: (Input<T> | undefined)[]): Output<Unwrap<T>[]>;
 export function all<T>(val: Input<T>[] | Record<string, Input<T>>): Output<any> {
-    if (val instanceof Array) {
-        const allOutputs = val.map(v => output(v));
-
-        const [syncResources, isKnown, isSecret, allResources] = getResourcesAndDetails(allOutputs);
-        const promisedArray = Promise.all(allOutputs.map(o => o.promise(/*withUnknowns*/ true)));
+    return output(val);
+}
 
-        return new Output<Unwrap<T>[]>(syncResources, promisedArray, isKnown, isSecret, allResources);
+function getAwaitableValue(v: any): any {
+    if (Output.isInstance(v)) {
+        return v.promise(/* withUnknowns */ true);
     } else {
-        const keysAndOutputs = Object.keys(val).map(key => ({ key, value: output(val[key]) }));
-        const allOutputs = keysAndOutputs.map(kvp => kvp.value);
-
-        const [syncResources, isKnown, isSecret, allResources] = getResourcesAndDetails(allOutputs);
-        const promisedObject = getPromisedObject(keysAndOutputs);
-
-        return new Output<Record<string, Unwrap<T>>>(syncResources, promisedObject, isKnown, isSecret, allResources);
+        return v;
     }
 }
 
 async function getPromisedObject<T>(
-        keysAndOutputs: { key: string, value: Output<Unwrap<T>> }[]): Promise<Record<string, Unwrap<T>>> {
+        keysAndOutputs: { key: string; value: any }[]): Promise<Record<string, Unwrap<T>>> {
     const result: Record<string, Unwrap<T>> = {};
     for (const kvp of keysAndOutputs) {
-        result[kvp.key] = await kvp.value.promise(/*withUnknowns*/ true);
+        result[kvp.key] = await getAwaitableValue(kvp.value);
     }
 
     return result;
 }
 
-function getResourcesAndDetails<T>(allOutputs: Output<Unwrap<T>>[]): [Set<Resource>, Promise<boolean>, Promise<boolean>, Promise<Set<Resource>>] {
+function getResourcesAndDetails(allValues: any[]): [Set<Resource>, Promise<boolean>, Promise<boolean>, Promise<Set<Resource>>] {
```

Comment:
In this and several other places, we lost some type checking - it seems that is roughly necessary - but it's a shame that we are losing some of the structure around the data flowing through these APIs.  Is there anything we can do to recover stronger typing of these signatures?

Reply:
can we not just weaken this to `Output<any>`?

Reply:
#Resolved